packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -0,0 +1,784 @@
+		const mapHandle = this.root.get<IFluidHandle<SharedMap>>(this.sharedMapKey);
+		assert(mapHandle !== undefined, "SharedMap not found");
+		this.map = await mapHandle.get();
+
+		const sharedCellHandle = this.root.get<IFluidHandle<SharedCell>>(this.sharedCellKey);
+		assert(sharedCellHandle !== undefined, "SharedCell not found");
+		this.sharedCell = await sharedCellHandle.get();
+	}
+}
+
+function getIdCompressor(dds: SharedObjectCore): IIdCompressor {
+	return (dds as any).runtime.idCompressor as IIdCompressor;
+}
+
+describeNoCompat("Runtime IdCompressor", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
+	const factory = new DataObjectFactory(
+		"TestDataObject",
+		TestDataObject,
+		[SharedMap.getFactory(), SharedCell.getFactory()],
+		[],
+	);
+
+	const runtimeOptions: IContainerRuntimeOptions = {
+		enableRuntimeIdCompressor: true,
+	};
+
+	const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+		factory,
+		[[factory.type, Promise.resolve(factory)]],
+		undefined,
+		undefined,
+		runtimeOptions,
+	);
+
+	let containerRuntime: ContainerRuntime;
+	let container1: IContainer;
+	let container2: IContainer;
+	let mainDataStore: TestDataObject;
+
+	let sharedMapContainer1: SharedMap;
+	let sharedMapContainer2: SharedMap;
+	let sharedMapContainer3: SharedMap;
+
+	let sharedCellContainer1: SharedCell;
+
+	const createContainer = async (): Promise<IContainer> =>
+		provider.createContainer(runtimeFactory);
+
+	beforeEach(async () => {
+		provider = getTestObjectProvider();
+		container1 = await createContainer();
+		mainDataStore = await requestFluidObject<TestDataObject>(container1, "/");
+		containerRuntime = mainDataStore._context.containerRuntime as ContainerRuntime;
+		sharedMapContainer1 = mainDataStore.map;
+		sharedCellContainer1 = mainDataStore.sharedCell;
+
+		container2 = await provider.loadContainer(runtimeFactory);
+		const container2MainDataStore = await requestFluidObject<TestDataObject>(container2, "/");
+		sharedMapContainer2 = container2MainDataStore.map;
+
+		const container3 = await provider.loadContainer(runtimeFactory);
+		const container3MainDataStore = await requestFluidObject<TestDataObject>(container3, "/");
+		sharedMapContainer3 = container3MainDataStore.map;
+
+		await waitForContainerConnection(container1);
+		await waitForContainerConnection(container2);
+		await waitForContainerConnection(container3);
+	});
+
+	it("has no compressor if not enabled", async () => {
+		provider.reset();
+		const config: ITestContainerConfig = {
+			registry: [
+				["mapId", SharedMap.getFactory()],
+				["cellId", SharedCell.getFactory()],
+			],
+			fluidDataObjectType: DataObjectFactoryType.Test,
+		};
+		const container = await provider.makeTestContainer(config);
+		const dataObject = await requestFluidObject<ITestFluidObject>(container, "default");
+		const map = await dataObject.getSharedObject<SharedMap>("mapId");
+
+		assert(getIdCompressor(map) === undefined);
```

Comment:
You should add a container loading with compressor on in container runtime options. Then validate that it is disabled. I believe it will fail, see - https://github.com/microsoft/FluidFramework/pull/14340#discussion_r1180629585.